### ISSUE TITLE
[Feature Request] Recommended dark mode for the current site

### ISSUE THREAD

**Issue by Gitoffthelawn** [darkreader/darkreader#6020]
# Feature Request

With the impressive recent improvements to Dynamic Mode, a majority of sites look good in that mode.

Yet even with the improvements, there are still some sites that look better with Static Mode, and some with Filter or Filter+ Modes.

Plus, although Dynamic Mode is *much, much, much* faster than before, it still is too slow for a few sites.

For the above reasons, I think it would be great for Dark Reader to have a "recommended mode" for sites.  This recommended mode could be saved in a community-editable database, much like the Dynamic Mode fixes database.

I haven't had the time to flush out my ideas of exactly how all this will work, but I'm thinking there could be an option that the user could enable that would allow Dark Reader to ***auto-select*** the mode based on this database.  Of course, a per-site override would still be offered.

Adding this useful functionality could be accomplished in the UI by simply adding an "Auto-Select" option to the `Mode` dropdown selector.  Then the user can select it as the "Theme for all websites".

**Comment by Gusted** [comment]
That means, we will have to maintain another list while I'm pro-actively trying to remove the dark-sites.config #5719 as these lists are a bit of a nightmare and waste of code. 

Also, I can always argue that dynamic is good not just because of performance, because in that case I we simply should remove the mode. But to make certain changes to the filtersettings. I'm never amused of these recommended settings. As everyone tend to change them along the way and at that point I don't really see why we should provide such thing.

**Comment by Gitoffthelawn** [comment]
@Gusted wrote:
> That means, we will have to maintain another list...

I agree.  I'm not a fan of lists or the need for either of the current exceptions files.  But the engine is currently not good enough to do without them, no?  I haven't run any metrics, but I would guess that about 5-10% of sites have text or other elements that are currently illegible with Dynamic Mode.  For Static Mode, that probably jumps to 15-25% (unless it has been significantly improved since the last time I tried to use it as my "main mode").  And Filter/Filter+ modes are obviously just invert CSS filters which make most text look real bad (and CSS filters, although I love them, are currently *very* processor intensive).

> Also, I can always argue that dynamic is good not just because of performance, because in that case I we simply should remove the mode. 

I agree that the performance of Dynamic Mode is not great, but IMHO it's the only thing that's better in Dark Reader than the [Dark Background and Light Text Extension](https://github.com/m-khvoinitsky/dark-background-light-text-extension), which I previously used.  DBLT's "static" mode is far superior than Dark Reader's static mode (IMO), but Dark Reader's Dynamic Mode is unparalleled (IMO).  I've actually considered seeing if it's possible to integrate the two into a single extension that would be the "best of both worlds". :)

> As everyone tend to change them along the way and at that point I don't really see why we should provide such thing.

That's the big issue, IMO.  As both Dynamic Mode and Static Mode improve on different trajectories, sometimes a site's "best mode" will change.  Ideally, Dark Reader could detect the best mode to use, but as you know, that's a big challenge.  So a static list to indicate the best mode is a stopgap measure just like both of the current "fixes" lists for Dynamic Mode and Filter/Filter+ Modes.  The community could be asked to see if anyone would be willing to volunteer to maintain the list, similar to how all the different filter lists are maintained for a project that relies on community lists like uBlock Origin.

**Comment by Gusted** [comment]
> But the engine is currently not good enough to do without them, no?

Yeah no. Dynamic mode is a generic mode nothing specific - none of the modes are specific, they are all generics. So they will assume/work on good amount and try to account for known edge cases. But certain cases requires little fixes. Like inverting images. Or some border that isn't being colored or some variable is being forgotten due some bug. 

>  The community could be asked to see if anyone would be willing to volunteer to maintain the list, similar to how all the different filter lists are maintained for a project that relies on community lists like uBlock Origin.

I don't see anyone doing it other then myself like the other configs at the moment and cannot be bother by some new repo that does this and we will have to pull from that one. 

I leave it open for now to get some more feedback - but for me I see it as a big hassle whereby users who will use it probably change it again and/or people will come and say "Oh yeah but on this specific part of the page it doesn't work" etc.

**Comment by arche-dev** [comment]
> Yet even with the improvements, there are still some sites that look better with Static Mode, and some with Filter or Filter+ Modes.

I have custom colors set for DR and if I'd visit a site that that recommends static mode and sets that, my colors will be ignored and it'll use some other color scheme, which I personally don't like
